examples/bond/test/bond-token-flow.js

```diff
@@ -0,0 +1,424 @@
+    * bob can exit all their tokens (12 and 2 respectively).
+    */
+
+    const currentBondIndex = runtime.addAsset(
+      'bond-token-0', { creator: { ...bondTokenCreator.account, name: 'bond-token-creator' } });
+
+    const creationFlags = Object.assign({}, flags);
+    const creationArgs = [
+      appManagerPk,
+      bondCreator,
+      issuePrice,
+      couponValue,
+      `int:${currentBondIndex}`,
+      maxIssuance
+    ];
+
+    // create application
+    applicationId = runtime.addApp(
+      { ...creationFlags, appArgs: creationArgs }, {}, approvalProgram, clearProgram);
+
+    // setup lsig account
+    // Initialize issuer lsig with bond-app ID
+    const scInitParam = {
+      TMPL_APPLICATION_ID: applicationId,
+      TMPL_OWNER: bondTokenCreator.address,
+      TMPL_APP_MANAGER: appManager.address
+    };
+    const issuerLsigProg = getProgram('issuer-lsig.py', scInitParam);
+    lsig = runtime.getLogicSig(issuerLsigProg, []);
+    issuerLsigAddress = lsig.address();
+
+    // sync escrow account
+    issuerAddress = runtime.getAccount(issuerLsigAddress);
+    console.log('Issuer Address: ', issuerLsigAddress);
+
+    // fund escrow with some minimum balance first
+    const fundEscrowParam = {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: master.account,
+      toAccountAddr: issuerLsigAddress,
+      amountMicroAlgos: minBalance + 10000,
+      payFlags: {}
+    };
+    runtime.executeTx(fundEscrowParam);
+
+    // verify global state
+    assert.isDefined(applicationId);
+    assert.deepEqual(getGlobal('app_manager'), convert.addressToPk(appManager.address));
+    assert.deepEqual(getGlobal('issue_price'), 1000n);
+    assert.deepEqual(getGlobal('coupon_value'), 20n);
+    assert.deepEqual(getGlobal('epoch'), 0n);
+    assert.deepEqual(getGlobal('current_bond'), BigInt(currentBondIndex));
+
+    // update application with correct issuer account address
+    const appArgs = ['str:update_issuer_address', convert.addressToPk(issuerLsigAddress)]; // converts algorand address to Uint8Array
+
+    const appCallParams = {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: appManager.account,
+      appID: applicationId,
+      payFlags: {},
+      appArgs: appArgs
+    };
+    runtime.executeTx(appCallParams);
+
+    // verify issuer address
+    assert.isDefined(applicationId);
+    assert.deepEqual(getGlobal('issuer_address'), convert.addressToPk(issuerLsigAddress));
+
+    // opt-in to app
+    runtime.optInToApp(appManager.address, applicationId, {}, {});
+    runtime.optInToApp(issuerAddress.address, applicationId, {}, {});
+
+    syncAccounts();
+    assert.isDefined(appManager.appsLocalState.get(applicationId));
+    assert.isDefined(issuerAddress.appsLocalState.get(applicationId));
+
+    runtime = optIn(runtime, lsig, currentBondIndex, appManager);
```

Comment:
I don't follow this. Could you please explain?

Reply:
could you tell me which part ? opt-in ?

Reply:
you mean assigning runtime ? yes we don't need that. or something else ?

Reply:
yes assigning runtime. Also can we rename `optIn`? Maybe `optInToBondApp`?

Reply:
this opt in is for lsig to bond. Renaming to optInLsigToBond